ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -749,7 +752,42 @@ getModSummaryFromImports env fp modTime contents = do
                 , ms_srcimps      = srcImports
                 , ms_textual_imps = textualImports
                 }
-    return (summary, allImps)
+
+    msrFingerprint <- liftIO $ computeFingerprint opts msrModSummary
+    return ModSummaryResult{..}
+    where
+        -- Compute a fingerprint from the contents of `ModSummary`,
+        -- eliding the timestamps, the preprocessed source and other non relevant fields
+        computeFingerprint opts ModSummary{..} = do
+            let moduleUniques =
+                    [ b
+                    | m <- moduleName ms_mod
+                         : map (unLoc . snd) (ms_srcimps ++ ms_textual_imps)
+                    , b <- toBytes $ uniq $ moduleNameFS m
+                    ] ++
+                    [ b
+                    | (Just p, _) <- ms_srcimps ++ ms_textual_imps
+                    , b <- toBytes $ uniq p
+                    ]
+            fingerPrintImports <- withArrayLen moduleUniques $ \len p ->
+                    fingerprintData p len
+            return $ fingerprintFingerprints $
+                    [ fingerprintString fp
+                    , fingerPrintImports
+                    ] ++ map fingerprintString opts
+
+        toBytes :: Int -> [Word8]
```

Comment:
Is there a way to avoid assuming 64-bit ints here?

Reply:
Good point, thanks